discussion/src/main/java/org/openedx/discussion/presentation/search/DiscussionSearchThreadViewModel.kt

```diff
@@ -84,26 +91,26 @@ class DiscussionSearchThreadViewModel(
             queryChannel
                 .asSharedFlow()
                 .debounce(400)
-                .collect {
+                .collect { query ->
                     nextPage = 1
-                    currentQuery = it
                     threadsList.clear()
-                    _uiState.value = DiscussionSearchThreadUIState.Loading
-                    loadThreadsInternal(currentQuery!!, nextPage!!)
+                    if (query.isNotEmpty()) {
+                        currentQuery = query
+                        _uiState.value = DiscussionSearchThreadUIState.Loading
+                        loadThreadsInternal(currentQuery!!, nextPage!!)
+
+                    } else {
+                        loadNextJob?.cancel()
+                        currentQuery = null
+                        _uiState.value = DiscussionSearchThreadUIState.Threads(emptyList(), 0)
```

Comment:
We also need to set `_canLoadMore.value = false` otherwise it will show the spinner as the last item on the list.

Reply:
done